src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java

```diff
@@ -112,6 +125,56 @@ public void deleteCard(Long cardId) {
     public boolean areThereNotPostponedCardsAvailable(Long deckId) throws NotAuthorisedUserException {
         User user = userService.getAuthorizedUser();
         return userCardQueueService.countCardsThatNeedRepeating(deckId) > 0 ||
-                cardRepository.getNewCards(deckId, user.getId(), user.getAccount().getCardsNumber()).size() > 0;
+                !cardRepository.getNewCards(deckId, user.getId(), user.getAccount().getCardsNumber()).isEmpty();
+    }
+
+    @Transactional
+    public void uploadCards(MultipartFile cardsFile, Long deckId) throws WrongFormatException, EmptyFileException, NotOwnerOperationException, NotAuthorisedUserException, IOException {
+        if (deckService.getDeckUser(deckId) != null) {
+            if (!cardsFile.getContentType().equals("application/octet-stream")) {
+                throw new WrongFormatException();
+            } else if (cardsFile.isEmpty()) {
+                throw new EmptyFileException("File is empty!");
+            }
+            Yaml yaml = new Yaml();
+            InputStream in = cardsFile.getInputStream();
+            try {
+                CardFileDTOList cards = yaml.loadAs(in, CardFileDTOList.class);
+                for (CardFileDTO card : cards.getCards()) {
+                    addCard(new Card(card.getQuestion(), card.getAnswer(), card.getTitle()), deckId);
+                }
+            } catch (ParserException | ConstructorException ex) {
+                throw new IllegalArgumentException("Invalid format of file!");
+            }
+        }
+    }
+
+    public void downloadCards(Long deckId, OutputStream outputStream) {
+        List<Map<String, String>> list = new ArrayList<>();
+        cardRepository.findAllByDeckId(deckId).forEach(card -> {
+            Map<String, String> cardMap = new HashMap<>();
+            cardMap.put("title", card.getTitle());
+            cardMap.put("question", card.getQuestion());
+            cardMap.put("answer", card.getAnswer());
+            list.add(cardMap);
+        });
+        DumperOptions options = new DumperOptions();
+        options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+        options.setDefaultScalarStyle(DumperOptions.ScalarStyle.SINGLE_QUOTED);
+        options.setPrettyFlow(true);
+        Map<String, List<Map<String, String>>> cardsMap = new HashMap<>();
+        cardsMap.put("cards", list);
```

Comment:
You can use [Collections#sinngletonMap](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#singletonMap-K-V-)

There are other useful builders in this class, e.g. Collection#emptyMap, just in case.